packages/website/ts/utils/utils.ts

```diff
@@ -313,14 +313,13 @@ export const utils = {
         const baseUrl = `https://${window.location.hostname}${hasPort ? `:${port}` : ''}`;
         return baseUrl;
     },
-    async onPageLoadAsync(): Promise<void> {
+    onPageLoadPromise: new Promise((resolve, _reject) => {
```

Comment:
Hmm this is a bit inconsistent with how we normally handle promises, is this functionally different than creating a new promise each time? Was the main issue that resolve wasn't being called in the if statement?

Reply:
The issue is that you can't set `window.onload = resolve` multiple times (ie. for multiple promises, a new one created for each function call) because window.onload can only be one function and is only called once. So with the previous method here is what would happen:
```
window.onload = A
window.onload = B
window.onload = C
window.onload() // C is called
```
The Promises made for A and B are never resolves and await forever.  Imho this is the best way to do it.

Reply:
I see, are we at risk of never resolving if we set `window.onload` too late?

Reply:
Previously, the only `onPageLoadAsync()` that worked was the last call, all previous Promises that were returned will never resolve because `window.onload` points to the latest resolve function. Calling `onPageLoadAsync()` after `window.onload` worked fine because we checked for that and returned.